lib/hammer_cli_foreman/ssh_keys.rb

```diff
@@ -0,0 +1,47 @@
+module HammerCLIForeman
+  class SSHKeys < HammerCLIForeman::Command
+    resource :ssh_keys
+    command_name "ssh-keys"
+    desc _("Manage SSH Keys sources.")
+
+    class ListCommand < HammerCLIForeman::ListCommand
+      output do
+        field :id, _("Id")
+        field :name, _("Name")
+        field :fingerprint, _("Fingerprint")
+        field :length, _("Length")
+        field :created_at, _("Created at"), Fields::Date
+      end
+
+      build_options
+    end
+
+    class CreateCommand < HammerCLIForeman::CreateCommand
+      command_name "add"
+
+      option "--key-file", "KEY_FILE", _("Path to a SSH public key"),
+             :format => HammerCLI::Options::Normalizers::File.new
+
+      def request_params
+        params = super
+        params['user_id'] ||= params['ssh_key']['user_id']
```

Comment:
This was needed to make it work with a real API, in the tests the the expectation of the `user_id` param gets satisfied, but when using the CLI the id gets wrapped into the ssh_key hash. I'm not sure if that is an issue to do with the API or the parameter resolving.